addon/services/breadcrumbs.js

```diff
@@ -15,6 +15,42 @@ export default Service.extend({
   init () {
     this._super(...arguments)
 
-    this.instances = A([])
+    this.containers = []
+  },
+
+  /**
+   * Methods
+   */
+
+  registerContainer (containerData) {
+    assert(
+      'An `element` is required to register a breadcrumb container',
+      containerData.element
+    )
+    assert(
+      'This container was already registered before',
+      !this.isContainerRegistered(containerData)
+    )
+
+    this.set('containers', [...this.containers, containerData])
+  },
+
+  unregisterContainer ({ element }) {
+    this.set(
+      'containers',
+      this.containers.filter(registeredContainer => {
+        return element !== registeredContainer.element
+      })
+    )
+  },
+
+  /* Private */
+
+  isContainerRegistered ({ element }) {
+    let containerData = this.containers.find(registeredContainer => {
```

Comment:
Could also use `some`? It already returns a Boolean => https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some. Not important though.

Reply:
And maybe find is more clear in this case anyways.

Reply:
Nah, some is better

Reply:
Forgot to add this one it seems?

Reply:
Yeaaaa, rolled back when debugging the test issue, and forgot to reapply it seems.

Reply:
Should be good good now